server/daemon.go

```diff
@@ -0,0 +1,69 @@
+/*
+ * UpdateHub
+ * Copyright (C) 2017
+ * O.S. Systems Sofware LTDA: contato@ossystems.com.br
+ *
+ * SPDX-License-Identifier:     GPL-2.0
+ */
+
+package server
+
+import (
+	"path"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+type Daemon struct {
+	fswatcher *fsnotify.Watcher
+	backend   *ServerBackend
+	done      chan bool
+}
+
+func NewDaemon(sb *ServerBackend) (*Daemon, error) {
+	d := &Daemon{
+		backend: sb,
+		done:    make(chan bool),
+	}
+
+	fswatcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	if err = fswatcher.Add(sb.path); err != nil {
+		return nil, err
+	}
+
+	d.fswatcher = fswatcher
+
+	return d, nil
+}
+
+func (d *Daemon) Run() {
+	go func() {
+		for {
+			select {
+			case event := <-d.fswatcher.Events:
+				if event.Op == fsnotify.Remove {
```

Comment:
I'm not sure if I understood this correctly. The "case" statement is for a "select" and not a "switch".